### PR TITLE
chore: move all env utils to env package

### PR DESCRIPTION
**Pull request by hezhizhen**
## Summary

1. Move all env related util functions to package `env`
2. Delete `cloud/envir`

### CODE REVIEW

**Comment by hezhizhen** [comment]
@savil

File: internal/impl/config.go [hunk 1]
```diff
@@ -106,7 +105,7 @@ func readConfigFromURL(url *url.URL) (*Config, error) {
 }
 
 func upgradeConfig(cfg *Config, absFilePath string) error {
-	if notUpgrade, _ := strconv.ParseBool(os.Getenv(env.DevboxDoNotUpgradeConfig)); notUpgrade {
+	if env.NotUpgradeConfig() {
```
Comment (savil) [inline internal/impl/config.go:108]: nit: `env.DoNotUpgradeConfig` reads more naturally

File: internal/env/util.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+func IsDevboxCloud() bool {
+	return os.Getenv(DevboxRegion) != ""
+}
+
+func IsDevboxShellEnabled() bool {
+	inDevboxShell, _ := strconv.ParseBool(os.Getenv(DevboxShellEnabled))
+	return inDevboxShell
+}
+
+func NotTrack() bool {
```
Comment (savil) [inline internal/env/util.go:22]: nit: `DoNotTrack` reads more naturally

**Review by savil — APPROVED** [review]

File: internal/env/util.go (hunk 2 above)
Comment (hezhizhen) [reply internal/env/util.go:22]: But the name is occupied by the env constant...

File: internal/impl/config.go (hunk 1 above)
Comment (hezhizhen) [reply internal/impl/config.go:108]: Okay, I'll update it in the next PR

File: internal/env/util.go (hunk 2 above)
Comment (hezhizhen) [reply internal/env/util.go:22]: I see. I'll create a PR to do so.

File: internal/impl/config.go (hunk 1 above)
Comment (hezhizhen) [reply internal/impl/config.go:108]: See https://github.com/jetpack-io/devbox/pull/954

File: internal/env/util.go (hunk 2 above)
Comment (hezhizhen) [reply internal/env/util.go:22]: See https://github.com/jetpack-io/devbox/pull/954

**Pull request merged**